cmd/launcher/doctor.go

```diff
@@ -0,0 +1,625 @@
+		controlServerURL = *flKolideServerURL
+
+	case *flKolideServerURL == "localhost:3443":
+		controlServerURL = *flKolideServerURL
+		// We don't plumb flRootPEM through to the control server, just disable TLS for now
+		insecureControlTLS = true
+
+	case *flKolideServerURL == "localhost:3000" || *flIAmBreakingEELicense:
+		controlServerURL = *flKolideServerURL
+		disableControlTLS = true
+	}
+
+	opts := &launcher.Options{
+		Autoupdate:                         *flAutoupdate,
+		AutoupdateInterval:                 *flAutoupdateInterval,
+		AutoupdateInitialDelay:             *flAutoupdateInitialDelay,
+		CertPins:                           certPins,
+		CompactDbMaxTx:                     *flCompactDbMaxTx,
+		Control:                            false,
+		ControlServerURL:                   controlServerURL,
+		ControlRequestInterval:             *flControlRequestInterval,
+		Debug:                              *flDebug,
+		DelayStart:                         *flDelayStart,
+		DisableControlTLS:                  disableControlTLS,
+		InsecureControlTLS:                 insecureControlTLS,
+		EnableInitialRunner:                *flInitialRunner,
+		EnrollSecret:                       *flEnrollSecret,
+		EnrollSecretPath:                   *flEnrollSecretPath,
+		AutoloadedExtensions:               flAutoloadedExtensions,
+		IAmBreakingEELicense:               *flIAmBreakingEELicense,
+		InsecureTLS:                        *flInsecureTLS,
+		InsecureTransport:                  *flInsecureTransport,
+		KolideHosted:                       *flKolideHosted,
+		KolideServerURL:                    *flKolideServerURL,
+		LogMaxBytesPerBatch:                *flLogMaxBytesPerBatch,
+		LoggingInterval:                    *flLoggingInterval,
+		MirrorServerURL:                    *flMirrorURL,
+		NotaryPrefix:                       *flNotaryPrefix,
+		NotaryServerURL:                    *flNotaryServerURL,
+		TufServerURL:                       *flTufServerURL,
+		OsqueryFlags:                       flOsqueryFlags,
+		OsqueryTlsConfigEndpoint:           *flOsqTlsConfig,
+		OsqueryTlsDistributedReadEndpoint:  *flOsqTlsDistRead,
+		OsqueryTlsDistributedWriteEndpoint: *flOsqTlsDistWrite,
+		OsqueryTlsEnrollEndpoint:           *flOsqTlsEnroll,
+		OsqueryTlsLoggerEndpoint:           *flOsqTlsLogger,
+		OsqueryVerbose:                     *flOsqueryVerbose,
+		OsquerydPath:                       osquerydPath,
+		RootDirectory:                      *flRootDirectory,
+		RootPEM:                            *flRootPEM,
+		Transport:                          *flTransport,
+		UpdateChannel:                      updateChannel,
+		UpdateDirectory:                    *flUpdateDirectory,
+	}
+
+	return opts, nil
+}
+
+func runCheckups(checkups []*checkup) {
+	failedCheckups := []*checkup{}
+
+	// Sequentially run all of the checkups
+	for _, c := range checkups {
+		err := c.run()
+		if err != nil {
+			failedCheckups = append(failedCheckups, c)
```

Comment:
Why build the array, and not just print here?

Reply:
I felt it was easier to quickly understand what checkups succeeded/failed if the summary is the last thing printed, rather than whoever is looking through this needing to scan through the whole file to find failed checkups.

Reply:
Okay!